src/components/MultiHopTrade/hooks/useAllowanceApproval/helpers.ts

```diff
@@ -63,12 +70,16 @@ export const getApprovalTxData = async (
   })
 
   const { networkFeeCryptoBaseUnit, ...fees } = await getFees({
-    accountNumber: tradeQuoteStep.accountNumber,
-    wallet,
     adapter,
     to: assetReference,
     value,
     data,
+    ...(from
+      ? {
+          from,
+          supportsEIP1559: supportsETH(wallet) && (await wallet.ethSupportsEIP1559()),
```

Comment:
it kills me inside that this is required 😂 surely we could do something like `await wallet.ethSupportsEIP1559?.()` where `ethSupportsEIP1559` is implicit false?

Reply:
Even better, `wallet` is actually properly narrowed down already, nice catch! df02f89b28